src/main/java/com/commercetools/sync/types/utils/TypeUpdateFieldDefinitionActionUtils.java

```diff
@@ -0,0 +1,277 @@
+package com.commercetools.sync.types.utils;
+
+import com.commercetools.sync.commons.exceptions.BuildUpdateActionException;
+import com.commercetools.sync.commons.exceptions.DuplicateKeyException;
+import com.commercetools.sync.commons.exceptions.DuplicateNameException;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.types.FieldDefinition;
+import io.sphere.sdk.types.Type;
+import io.sphere.sdk.types.commands.updateactions.AddFieldDefinition;
+import io.sphere.sdk.types.commands.updateactions.ChangeFieldDefinitionOrder;
+import io.sphere.sdk.types.commands.updateactions.RemoveFieldDefinition;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.commercetools.sync.commons.utils.CommonTypeUpdateActionUtils.buildUpdateAction;
+import static com.commercetools.sync.types.utils.FieldDefinitionUpdateActionUtils.buildActions;
+import static java.lang.String.format;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+public final class TypeUpdateFieldDefinitionActionUtils {
+
+    /**
+     * Compares a list of {@link FieldDefinition}s with a list of {@link FieldDefinition}s.
+     * The method serves as a generic implementation for field definitions syncing and building the required
```

Comment:
why is it a generic implementation? I think it was copied from another doc from asset syncing. But it doesn't apply in this context 😉